src/bp-search/plugins/search-cpt/class-bp-search-cpt.php

```diff
@@ -29,6 +29,15 @@ public function __construct( $cpt_name, $search_type ) {
 			$this->search_type = $search_type;
 
 			add_action( "bp_search_settings_item_{$this->cpt_name}", array( $this, 'print_search_options' ) );
+			add_filter( "parse_query", array( $this, 'parse_query' ) );
+		}
+
+		// To bypass "The Event Calendar" plugin's parse_query filter
+		public function parse_query( $query ){
+			if( is_search() || ( defined('DOING_AJAX') && DOING_AJAX && $_REQUEST['action'] == 'bp_search_ajax' ) )  {
+				$query->set( 'tribe_suppress_query_filters', true );
+			}
+			return $query;
 		}
```

Comment:
@ngolwalkar This is about the third-party plugin compatibility and we have a file here `src/bp-core/compatibility/bp-incompatible-plugins-helper.php` so this code should be in `/bp-incompatible-plugins-helper.php` instead of here and run the PHPCS and fix all the errors.